packages/squiggle-lang/src/value/domain.ts

```diff
@@ -0,0 +1,55 @@
+import { ErrorMessage, REOther } from "../errors/messages.js";
+import { Err, Ok, result } from "../utility/result.js";
+import { Value } from "./index.js";
+
+abstract class BaseDomain {
+  abstract type: string;
+
+  abstract toString(): string;
+
+  abstract includes(value: Value): boolean;
+}
+
+class NumericRangeDomain extends BaseDomain {
+  readonly type = "NumericRange";
+
+  constructor(public min: number, public max: number) {
+    super();
+  }
+
+  toString() {
+    return `Range(${this.min} to ${this.max})`;
```

Comment:
Maybe, ``NumericRange({min: 5, max:10})``. This also highlights that you can do "foo.min"